server/api.js

```diff
@@ -2,100 +2,416 @@ import { Router } from "express";
+			[plan_id]
+		);
+
+		const result = await pool.query(
+			"DELETE FROM plans WHERE plans.id=$1 AND user_id=$2",
+			[plan_id, user_id]
+		);
+
+		res.json({ Success: "Plan has been deleted" });
+	} catch (err) {
+		res.status(500).send("server error");
+	}
+});
+
+// GET goals form specific user id
+router.get("/users/:user_id/goals", (req, res) => {
+	const { user_id } = req.params;
+	const query = `SELECT p.id, p.user_id, p.description,ARRAY_AGG(g.title  || ' status:'|| g.status) goals_list
+					FROM goals g INNER JOIN plans p on p.id=g.plan_id
+					WHERE p.user_id=$1 GROUP BY p.id, p.user_id, p.description`;
+
+	pool
+		.query(query, [user_id])
+		.then((result) => {
+			if (result.rowCount) {
+				res.json(result.rows);
+			} else {
+				res.status(404).json({
+					status: 404,
+					msg: "User doesn't have any goals.",
+				});
+			}
+		})
+		.catch((err) => console.error(err.message));
+});
+
+// GET a goal form specific plan id
+router.get("/users/:user_id/plans/:plan_id/goals", (req, res) => {
+	const { user_id, plan_id } = req.params;
+	const query = `SELECT p.user_id, p.id plan_id, g.id goal_id,  g.title, g.status, g.start_date, g.end_date, g.create_date  FROM goals g 
+						INNER JOIN plans p on p.id = g.plan_id where p.user_id = $1 and g.plan_id=$2`;
+	pool
+		.query(query, [user_id, plan_id])
+		.then((result) => {
+			if (result.rowCount) {
+				res.json(result.rows);
+			} else {
+				res.status(404).json({
+					status: 404,
+					msg: "This user doesn't have a goal with this id",
+				});
+			}
+		})
+		.catch((err) => console.error(err.message));
+});
+
+//POST a goal for specific plan id
+router.post("/users/plans/:plan_id/goals", (req, res) => {
+	const { plan_id } = req.params;
+	const { title, status, start_date, end_date } = req.body;
+	const query =
+		"INSERT INTO goals ( plan_id, title, status, start_date, end_date ) VALUES ($1, $2, $3, $4, $5) RETURNING *";
+	pool
+		.query(query, [plan_id, title, status, start_date, end_date])
+		.then((result) => {
+			res.json({ success: "New plan has been created!", user: result.rows[0] });
+		})
+		.catch((err) => console.error(err.message));
+});
+
+//DELETE a goal for specific plan id
+router.delete("/users/plans/:plan_id/goals/:goal_id", (req, res) => {
+	const { plan_id, goal_id } = req.params;
+	pool
+		.query("SELECT * FROM goals WHERE plan_id = $1 and id = $2", [
+			plan_id,
+			goal_id,
+		])
+		.then(() => {
+			pool
+				.query("DELETE FROM goals WHERE id=$1", [goal_id])
+				.then(() => res.send(`goal ${goal_id} has been deleted!`))
+				.catch((e) => console.error(e));
+		})
+		.catch((err) => console.error(err.message));
+});
+
+// GET all tasks form specific goal id
+router.get(
+	"/users/:user_id/plans/:plan_id/goals/:goal_id/tasks",
+	(req, res) => {
+		const { user_id, plan_id, goal_id } = req.params;
+		const query = `SELECT t.id, t.goal_id goal_id, t.description, t.status, t.create_date  FROM tasks t 
+						INNER JOIN goals g on t.goal_id = g.id 
```

Comment:
This query can be simplified.
You only need the task data and you have the goal_id already stored on the task table, so why do all the JOINS?

I think this is better designed than the goals endpoint above because you return only task-related data here. Maybe it's worth revisting the goals enpoint having this in mind